compiler/passes/convert-typed-uast.cpp

```diff
@@ -4938,8 +4938,32 @@ Expr* TConverter::ActualConverter::convertActualWithArg(const FormalActual& fa)
             astActual->toStringLiteral()->value().c_str());
         temp = new SymExpr(sym);
       } else {
-        auto type = tc_->convertType(fa.formalType().type());
-        temp = tc_->storeInTempIfNeeded(new CallExpr(PRIM_CAST, type->symbol, temp), fa.formalType());
+        auto kind = got.conversionKind();
+        if (kind == CanPassResult::ConversionKind::BORROWS ||
+            kind == CanPassResult::ConversionKind::BORROWS_SUBTYPE) {
+          auto ct = fa.actualType().type()->toClassType();
+          Expr* borrow = nullptr;
+          if (ct->manager()) {
+            auto mrt = types::QualifiedType(fa.actualType().kind(),
+                                            ct->managerRecordType(context));
+            types::QualifiedType qtField;
+            // Can always generate an access for 'chpl_p', since we're either accessing
+            // that field directly, or forwarding to it.
+            borrow = codegenGetFieldImpl(tc_, PRIM_UNKNOWN, temp, mrt,
```

Comment:
Do we have to call the impl here? If there's something that's missing from the overloads, can we adjust them or add a new one?

Reply:
We call the impl here because the AST has the type of the class, rather than the _owned/_shared record.